File: 프로그래머스/lv0/120813. 짝수는 싫어요/짝수는 싫어요.js
function solution(n) {
    let result = [];
    for(let i=0; i<=n; i++) {
        if(i%2 === 1) {
            result.push(i);
        }
    }
    return result;
}
/*
    접근법 :
    1. 0~n 개까지 순회 하면서 조건문을 달아 홀수만 배열에 저장
    2. 홀수만 들어갈 배열 선언
    3. n개까지 순회하는 for문 선언
    4. for문 안 홀수만 판별해내는 if문 선언후 통과되면 배열에 추가

    다른 풀이 :
    1. result.filter( (value) => { result value%2 === 1; } )
        -> 내장함수( .filter() ) 사용
    2. (int i=1; i<=n; i+=2)
        -> for문의 조건식 만으로 홀수만 집어넣기

*/

File: 프로그래머스/lv0/120813. 짝수는 싫어요/짝수는 싫어요.java
import java.util.ArrayList;
class Solution {
    public ArrayList<Integer> solution(int n) {
        ArrayList<Integer> list=new ArrayList<Integer>(); 
        
        for(int i=1; i<=n; i+=2) {                        
            list.add(i);                                
        }
        
        return list;
    }
}
/*
    접근법 :
    그냥 배열로 풀어보려다 방법이 떠오르지 않아 크기가 가변적인 ArrayList를 사용
    1. ArrayList를 사용하기 위해 임포트 구문
    2. 메소드의 반환타입 맞춰주기
    3. 메소드 안에 ArrayList 객체 생성해주기 -> list
    4. for문의 조건식 만으로 list에 홀수만 넣기 -> (int i=1; i<=n; i+=2)
    5. list 반환
*/
